### PR TITLE
Ask for help button feature

### LINKED ISSUE #15669: "Ask for help" button that pre-populates forum post
# User story
As a user, if I need help with a challenge, I can click a "Ask for help on the forum" button. A modal will pop up with the following text:

"If you've already tried [the Read-Search-Ask Method](https://forum.freecodecamp.org/t/the-read-search-ask-methodology-for-getting-unstuck/137307), then you can ask for help on the freeCodeCamp forum." [Create a help post on the forum]

A new tab will open with a prep-populated forum post with: 
- the the title of my current challenge in the subject line
- the post body, which includes this:

#### My code so far
*this should be automatically populated and formatted*

#### My browser information

User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36</code>.

# Discourse URL

Here's how to format a URL for creating a post on Discourse

`http://forum.freecodecamp.org/new-topic?title=**topic%20title**&body=**topic%20body**&category=help`

# UI element

Here's what the button should look like. This should be added to all views that have tests:

### PR DESCRIPTION
**Pull request by prateekgoel**
#### Pre-Submission Checklist

- [ ] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [ ] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue: Closes #15669

#### Description

Implemented the 'Ask for help' button feature for production.

### CODE REVIEW

File: server/views/partials/challenge-modals.jade [hunk 1]
```diff
@@ -19,3 +19,16 @@
...
+    .modal-dialog.animated.fadeInUp.fast-animation
+        .modal-content
+            .modal-header.challenge-list-header Ask for help?
+                a.close.closing-x(href='#', data-dismiss='modal', aria-hidden='true') ×
+            .modal-body
+                h3
+                  | If you've already tried the&nbsp;
+                  a(href='https://forum.freecodecamp.org/t/the-read-search-ask-methodology-for-getting-unstuck/137307', target='_blank') Read-Search-Ask&nbsp;
+                  | Method, then you can ask for help on the freeCodeCamp forum.
+                a.btn.btn-lg.btn-primary.btn-block#ask-for-help-button(name='_csrf', value=_csrf) Create a help post on the forum
```
Comment (piotrek1543) [inline server/views/partials/challenge-modals.jade]: I think that `Create a help post` is enough for link text and `on the forum` might be omitted as it is already explained in description. For me it seems like doubled content.

Comment (prateekgoel) [inline server/views/partials/challenge-modals.jade]: @piotrek1543 I kept it as descibed in the issue by @QuincyLarson .
However, if you want I will make it short.

Comment (QuincyLarson) [inline server/views/partials/challenge-modals.jade]: @piotrek1543 I think "ask for help on the forum" will be more clear to newcomers.

**Comment by QuincyLarson** [comment]
@prateekgoel This works great! Could you also add a **link to the challenge** to the end of post body? This will save people who are trying to help campers a lot of time.

**Comment by prateekgoel** [comment]
@QuincyLarson Sure. I will add the link too.

**Comment by prateekgoel** [comment]
@QuincyLarson Added the link :)

**Review by ghost — APPROVED** [review]

**Comment by QuincyLarson** [comment]
@prateekgoel I'm in the process of QA'ing this. Thanks for your patience.

**Comment by prateekgoel** [comment]
@QuincyLarson Ok thanks. 😃

**Comment by QuincyLarson** [comment]
@prateekgoel Awesome! This looks good to me. Waiting for @BerkeleyTrue to approve it as well, since this will go straight into production.

**Comment by prateekgoel** [comment]
@QuincyLarson Thanks. I am happy I could help.

File: client/commonFramework/ask-for-help.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        window.location.href + '\n'
+
+      ].join('');
+
+      helpTitle = encodeURIComponent(helpTitle);
+      helpMessage = encodeURIComponent(helpMessage);
+
+      $('#ask-for-help-modal').modal('hide');
+      window.open(
+        'http://forum.freecodecamp.org/new-topic?title='
```
Comment (BerkeleyTrue) [inline client/commonFramework/ask-for-help.js]: This should either be https or omitted

**Review by BerkeleyTrue — CHANGES_REQUESTED** [review]

File: client/commonFramework/ask-for-help.js (hunk 2 above)
Comment (prateekgoel) [inline client/commonFramework/ask-for-help.js]: @BerkeleyTrue Yes I missed that! I will change it to https.

**Comment by QuincyLarson** [comment]
@BerkeleyTrue It looks like @prateekgoel made the changes. Can you take look, then merge and deploy?

**Review by BerkeleyTrue — APPROVED** [review]

**Comment by prateekgoel** [comment]
@QuincyLarson @BerkeleyTrue I am not able to see 'Ask for help' button on freeCodeCamp site. What might be the reason?

**Comment by QuincyLarson** [comment]
@prateekgoel We're waiting for @BerkeleyTrue to deploy it.

**Comment by prateekgoel** [comment]
Ok Thanks @QuincyLarson . I hope it gets deployed soon.

**Comment by BerkeleyTrue** [comment]
This is now live. Thanks for your patience. Happy Coding.

**Comment by Zou1738** [comment]
I need theses messages to STOP

**Pull request merged**